src/toc.ts

```diff
@@ -23,12 +33,14 @@ export function activate(context: ExtensionContext) {
 }
 
 async function createToc() {
-    let editor = window.activeTextEditor;
+    const editor = vscode.window.activeTextEditor;
 
-    if (!isMdEditor(editor)) {
+    if (!(editor && isMdEditor(editor))) {
```

Comment:
`isMdEditor` will do the null check. I think it is reasonable (`null` is not a Markdown editor of course) and is also quite handy (don't need to manually check it every time).

Reply:
IMHO, a function is a **black box** to the caller.

The return value of `(editor): boolean` is `true` does **not** mean `editor` is not null.

Reply:
I don't quite understand 😳.

`isMdEditor` returns `true` only when it is a Markdown editor (allowing `null` as input).

https://github.com/yzhang-gh/vscode-markdown/blob/eff0d51e5c271c9fafaa3dfff144cc7c7ae4e7da/src/util.ts#L16-L18

Reply:
I mean, you cannot prove such inference:

> Let,
>
> * `e` be a `nullable` object.
> * `f` be a `(p: object) => boolean` function.
>
> If,
>
> * `f(e)` is `true`.
>
> Then,
>
> * `e` is not `null`.

---

See, in C#, we write a lot of

```csharp
if (paramA == null)
{
    throw new ArgumentNullException(nameof(paramA));
}
```

C# is a language, which requires **strict static type checks**.

**TypeScript is a language, which adds such checks to JavaScript.**

I believe we should adhere to the type system.

Reply:
Why do I need to think of any possible `f` 😳? Here we know (in our implementation) if `isMdEditor(e)` is `true`, then `e` is not `null`. And it is logically sound from the name.

Reply:
**strict static type checks**

Compiler does not care about the meaning of the function name. And here is no easy way to tell you intention to it.

Reply:
I get it now. (Again, I am so used to dynamic programming languages and didn't have the context. It could have been better if you reminded me in the beginning that we(/you) are adding strict type check. 😓)